tests/unit/test_main.py

```
@@ -553,3 +569,246 @@ def test_integration_complete_workflow():
+    mock_result.stderr = ''
+
+    # Mock provider
+    mock_provider = Mock()
+    mock_provider.is_available.return_value = True
+    mock_fallback_result = Mock()
+    mock_fallback_result.provider = mock_provider
+    mock_fallback_result.provider_name = 'ollama'
+    mock_fallback_result.had_fallback = True
+
+    # Track callback invocations
+    callback_calls = []
```

Ruff (0.14.10)

Local variable `callback_calls` is assigned to but never used

Remove assignment to unused variable `callback_calls`

(F841)